packages/data-stores/src/plans/selectors.ts

```diff
@@ -56,6 +56,24 @@ export const getPlanProductById = (
 		.find( ( product: PlanProduct ) => product.productId === productId );
 };
 
+export const getPlanProductByStoreSlug = (
+	_state: State,
+	planSlug: string | undefined,
+	locale: string
+): PlanProduct | undefined => {
+	if ( ! planSlug ) {
+		return undefined;
+	}
+
+	// This is a temporary workaround / sanity-fallback to ensure plans data
+	// are fetched when the selector is called. It should be refactored and removed.
+	select( STORE_KEY ).getSupportedPlans( locale );
```

Comment:
This will need rethinking/updating, but it works and it's more for sanity in case the selector is used outside of the connected components above (which would already reach the resolver).

Reply:
I know this is temporary, but, can we possibly check whether the plans are already available before calling this function?